service/lib/dinstaller/dbus/software/proposal.rb

```diff
@@ -64,18 +66,28 @@ def initialize(logger)
           dbus_method :SetResolvables,
             "in Id:s, in Type:y, in Resolvables:as, in Optional:b" do |id, type, resolvables, opt|
             Yast::PackagesProposal.SetResolvables(id, TYPES[type], resolvables, optional: opt)
+            notify_change!
           end
 
           dbus_method :RemoveResolvables,
             "in Id:s, in Type:y, in Resolvables:as, in Optional:b" do |id, type, resolvables, opt|
             Yast::PackagesProposal.RemoveResolvables(id, TYPES[type], resolvables, optional: opt)
+            notify_change!
           end
         end
 
+        def on_change(&block)
+          @on_change_callbacks << block
+        end
+
       private
 
         # @return [Logger]
         attr_reader :logger
+
+        def notify_change!
```

Comment:
NP: Why a bump method?

Reply:
see discussion at https://github.com/yast/d-installer/pull/381#discussion_r1059424403

Reply:
I like what is suggested here https://thecodest.co/blog/ruby-software-development-what-are-the-bang-methods-and-when-to-create-them/: "There is no point in creating a bang method if its counterpart without an exclamation mark does not exist. There are methods that are dangerous because they change the object or raise an exception, and yet they don't have an exclamation mark at the end of the name. Creating a bang method with no equivalent would confuse a programmer using your code."

Reply:
It is completely fine for me if we use it consistently :)